output/tar/tar_output.go

```diff
@@ -51,10 +51,9 @@ func (i Output) Apply(rootPath string) <-chan error {
 
 		err = tar.Run()
 		if err != nil {
-			done <- err
+			done <- output.Report{output.UnknownError.Wrap(err).(*errors.Error), def.Output{}}
 			return
 		}
-
 	}()
 	return done
```

Comment:
So, two separate thoughts.

First, I had forgotten that we close the channel and that was previously our de-facto "report". While I had thought this was totally broken, it's actually more subtle than that.

In `provision.go`, you check for `report.Err != nil` where `report` is the result of a channel read - what is the behaviour of that check when the channel was closed without being written to? I would assume either NPE, or having a "fake" default report struct.

Second, I think it's a pretty safe assumption that exec-wrapping transports will be around for awhile, and IMO they should never go away entirely. I can envision way too many cases where our nascent transports will want to be ditched (hopefully temporarily) in the name of pragmatism, and will thus need to always ensure it's possible to Accomplish Real Things on master. This isn't anything new, just our ongoing efforts to reconcile a useful execution platform with goals of repeatable computing. 

Along these lines, I think it'd be very reasonable to expect that the `output.Apply()` interface method is expected to always Report on its channel with whatever information it has. Which is this case means adding a Report in the success case of a nil Error and an Output that is a byvalue-copy of its call parameter, with the Hash field explicitly set to `""`.